### DISCUSSION TITLE
How to use CIGALE2022 for fitting FIR components well?: differences from X-CIGALE (=> How to treat no data, upper limit and additional error size)

### DISCUSSION THREAD

**Discussion by HikaruF9** [mboquien/cigale#8] [Q&A] [answered]
Hi,
I have tried the SED fitting of dust obscured galaxies. Previously I used X-CIGALE, but I am trying the new version CIGALE2022.0.  

However, I cannot obtain a good fitting result with the same parameter set with X-CIGALE.  
I attached the SED fitting results of a source both using X-CIGALE (top figure) and CIGALE2022.0 (bottom figure).
![X_CIGALE_best_fit](https://user-images.githubusercontent.com/98146230/164616864-802ea01b-c10d-4fc4-8b95-77f39f088cdb.png)
![CIGALE2022_best_fit2](https://user-images.githubusercontent.com/98146230/164617478-a9162af0-3e1e-4bf5-958b-841fbbba1e54.png)

Far infra-red emission can not reach the observed flux in CIGALE2022, and this trend is also seen in other sources.
Could you please check the figure and the problem?
Here is the photometry data for this source and pcigale.ini (txt. version).
[obj1_sed.txt](https://github.com/mboquien/cigale/files/8537774/obj1_sed.txt)
[pcigale.ini.txt](https://github.com/mboquien/cigale/files/8537779/pcigale.ini.txt)

**Comment by HikaruF9** [comment, marked as answer]
When I remove the AKARI data (upper limit), the fitting becomes well.


![CIGALE2022_without_AKARI](https://user-images.githubusercontent.com/98146230/164672946-19f6e115-321a-43e9-832b-bef09b0e6ce4.png)


I obtain good fitting results also for other sources without AKARI data. I found that if I do fitting with AKARI data in X-CIGALE, the fitting result becomes bad. 
It seems that the AKARI filter or some thing is wrong..

**Reply by mboquien** [reply]
This is extremely curious. The best-fit clearly does not correspond to the displayed χ². The issue comes from the Wide-S band? It seems the run does not correspond exactly to the input file (N60 and Wide-L have fluxes at -1). Could you provide the exact flux file and also the pcigale.ini.spec? I would like to give it a try.

**Reply by Guang91** [reply]
After playing with your data/configurations, I think the issue is your data format. As @mboquien says, the N60 and Wide-L fluxes are set to -1 (I guess they're invalid data right?). With this setting, the new code considers -1 as the flux upper limit, which messes up the fit. We suggest setting invalid data to nan instead of a specific number. By doing so, the fit becomes good as below
![image](https://user-images.githubusercontent.com/7922328/164770284-d6b8976f-8cf4-4254-9457-991ed6953746.png)

**Reply by HikaruF9** [reply]
Oh, I see. I followed X-CIGALE manual, but will change -1 to nan for no data (e.g., H-ATLAS, VISTA(VIKING) for some sources).

When I input nan for the columns, below error occur.
"The input could not be read properly. Verify its format and that it does not have two columns 
with the same name."
It seems 'nan' is not read, and the number of data becomes smaller than columns.

How can I treat nan?

**Reply by Guang91** [reply]
The manual is a bit outdated, sorry. We should update the manual. 

Do you mean CIGALE v2022.0 doesn't read your new (nan) data? 
It's strange. Could you compare your data file vs. the attached (which works well on my machine)?

[obj1_sed.zip](https://github.com/mboquien/cigale/files/8551283/obj1_sed.zip)

**Reply by HikaruF9** [reply]
I compared with your data file.
I found that my data is wrong, and I should input 'nan' as string format. 
Now I can succeed to fit well. Thank you so much for your help.

**Comment by Guang91** [comment]
@mboquien Shall we explain the data format somewhere? Many users still follow the old manual ("CIGALE-v0.9.0-HowTo.pdf", I also suggest updating this manual). 
Specifically, I mean the following figure from the old manual, which is really misleading. 
![image](https://user-images.githubusercontent.com/7922328/164771046-e94efb33-be00-4fa9-b062-b6bbd2d200f4.png)

**Reply by HikaruF9** [reply]
May I ask you about a related question? I would like to know how to change the value of the additional error. 

It seems the default error = 10% for the additional error, and it is set in observation.py. 
However I want to check the result with 0% additional error or 20% etc.

**Reply by mboquien** [reply]
If you have the latest git version of cigale (develop branch), you can change it from the pcigale.ini file. If not, you can change `modelerror` on this line https://gitlab.lam.fr/cigale/cigale/-/blob/master/pcigale/managers/observations.py#L35 .

**Reply by HikaruF9** [reply]
Thank you so much for this quick reply. 
I am now able to change the error size.

**Reply by belenalk** [reply]
I have a related question. When using upper limits (flux >0 and error =-99.0 and lim_flag = True in cigale 2020.0) they are not used as real upper limits. I mean, cigale applies an error  bar that goes up to 100mJy. Is there a way to force cigale to only consider models that go below those fluxes? And also has that behaviour changed in cigale 2022 version? Thanks in advance!

**Reply by mboquien** [reply]
The error is the uncertainty on the upper limit. It is not supposed to be -99 (unless it has been observationally determined to be ±99 mJy). If it is unknown, the error could be taken as minus the upper limit.

**Reply by belenalk** [reply]
Ok, I see.  Thanks a lot for such a quick answer!
